ts/screens/authentication/IdpLoginScreen.tsx

```diff
@@ -111,127 +107,166 @@ const styles = StyleSheet.create({
-    this.setState({ loginTrace: url });
+  const updateLoginTrace = (url: string): void => {
+    setLoginTrace(url);
   };
 
-  private handleLoadingError = (error: WebViewErrorEvent): void => {
+  const handleLoadingError = (error: WebViewErrorEvent): void => {
     const { code, description, domain } = error.nativeEvent;
     void mixpanelTrack("SPID_ERROR", {
-      idp: this.props.loggedOutWithIdpAuth?.idp.id,
+      idp: props.loggedOutWithIdpAuth?.idp.id,
       code,
       description,
       domain
     });
-    this.setState({
-      requestState: pot.noneError(ErrorType.LOADING_ERROR)
-    });
-  };
 
-  private handleLoginFailure = (errorCode?: string) => {
-    this.props.dispatchLoginFailure(
-      new Error(`login failure with code ${errorCode || "n/a"}`),
-      this.idp
-    );
-    const logText = pipe(
-      errorCode,
-      O.fromNullable,
-      O.fold(
-        () => "login failed with no error code available",
-        ec =>
-          `login failed with code (${ec}) : ${getSpidErrorCodeDescription(ec)}`
-      )
-    );
-
-    handleSendAssistanceLog(this.choosenTool, logText);
-    this.setState({
-      requestState: pot.noneError(ErrorType.LOGIN_ERROR),
-      errorCode
-    });
+    setRequestState(pot.noneError(ErrorType.LOADING_ERROR));
   };
 
-  private handleLoginSuccess = (token: SessionToken) => {
-    handleSendAssistanceLog(this.choosenTool, `login success`);
-    this.props.dispatchLoginSuccess(token, this.idp);
+  const handleLoginFailure = useCallback(
+    (code?: string) => {
+      props.dispatchLoginFailure(
+        new Error(`login failure with code ${code || "n/a"}`),
+        idp
+      );
+      const logText = pipe(
+        code,
+        O.fromNullable,
+        O.fold(
+          () => "login failed with no error code available",
+          ec =>
+            `login failed with code (${ec}) : ${getSpidErrorCodeDescription(
+              ec
+            )}`
+        )
+      );
+
+      handleSendAssistanceLog(choosenTool, logText);
+      setRequestState(pot.noneError(ErrorType.LOGIN_ERROR));
+      setErrorCode(code);
+    },
+    [choosenTool, idp, props]
+  );
+
+  const handleLoginSuccess = (token: SessionToken) => {
+    handleSendAssistanceLog(choosenTool, `login success`);
+    props.dispatchLoginSuccess(token, idp);
   };
 
-  private setRequestStateToLoading = (): void =>
-    this.setState({ requestState: pot.noneLoading });
+  const setRequestStateToLoading = (): void => setRequestState(pot.noneLoading);
 
-  private handleNavigationStateChange = (event: WebViewNavigation): void => {
-    if (event.url) {
-      const urlChanged = getUrlBasepath(event.url);
-      if (urlChanged !== this.state.loginTrace) {
-        this.props.dispatchIdpLoginUrlChanged(urlChanged);
-        this.updateLoginTrace(urlChanged);
+  const handleNavigationStateChange = useCallback(
+    (event: WebViewNavigation) => {
+      if (event.url) {
+        const urlChanged = getUrlBasepath(event.url);
```

Comment:
`urlChanged` does not reflect the true meaning of the constant, should we use something more meaningful, like `urlBasePath`?